server/game/gameSystems/MoveCardSystem.ts

```diff
@@ -1,11 +1,12 @@
 import type { AbilityContext } from '../core/ability/AbilityContext';
 import type { Card } from '../core/card/Card';
-import { CardType, EffectName, EventName, Location, WildcardCardType } from '../core/Constants';
+import { CardType, EffectName, EventName, GameStateChangeRequired, Location, LocationFilter, WildcardCardType } from '../core/Constants';
 import * as EnumHelpers from '../core/utils/EnumHelpers';
 import { type ICardTargetSystemProperties, CardTargetSystem } from '../core/gameSystem/CardTargetSystem';
 
 export interface IMoveCardProperties extends ICardTargetSystemProperties {
     destination?: Location;
```

Comment:
I think it would be good to constrain the acceptable values for location since there are more specific systems that should be used in some cases. Specifically, this should never be used to move a card into discard, resources, or play arenas as we have more specific systems for those.

Probably would be good to add short jsdoc on the property and on the system itself briefly explaining why it's set up that way as well.

Reply:
`PlayEventAction` uses `MoveCardSystem` to move cards to the discard so we won't be able to remove that location.